### ISSUE TITLE
Give a warning before some missions are aborted

### ISSUE THREAD

**Issue by jafdy** [endless-sky/endless-sky#2640]
I have noticed an increase in people who have aborted important story missions because they couldn't do them. 
This may not be needed once the full set of main campaigns are finished.

Drone Hunting (FW Katya 3) Could do with this kind of warning more than most as it is too easy to cause the story to fail at this point.

**Comment by Amazinite** [comment]
For reference: http://steamcommunity.com/app/404410/discussions/0/2333276539596843266/

**Comment by jafdy** [comment]
Yes, that's precisely what I'm thinking. Although if someone could find a less 4th wall breaking way of putting it that would be great.

**Comment by endless-sky** [comment]
Once you commit to the Free Worlds, there are warnings if you fail or abort a mission in the main story line. But I feel like adding those warnings before you've committed would sound too much like the game is trying to force you to do something. Part of the point of an open world game, after all, is that it doesn't hold your hand and force you to go in a certain direction.

**Comment by jafdy** [comment]
A lot of people abort thinking they can come back to it later. So a warning saying "If you abort now you won't be able to come back to this later." is not going to force people too much. It's telling you that if you do what your about to do then a path will be shut off permanently.

**Comment by tehhowch** [comment]
Such a warning could be removed when alternate ways of advancing the Main Plot are available. As it is, if you cannot start or continue the Free Worlds plot, then you are also unable to experience the full Wanderer plot.

**Comment by Forlarren** [comment]
This is really a mission design issue. 

There is no reason to have fail states. Well there are, but they are bad reasons.

Modern table tops fail forward concept should be adapted by everyone. It's simple, failure should lead to another conflict (synonymous with opportunity) that leads to success.

Lets say you fail some sneaky mission, well instead of reloading you now have to disable and board the ship that took it from you, a ship that just happens to be the perfect ship for the next mission in the series, so if you manage to capture it, it's a big boon.

Many missions just need a "try again" option, this time it's harder and the player must now grind since the enemy is dug in and prepared for them. Again if the player can defeat the upgraded force next mission they will be ahead of the curve again.

Since there is a big clock of stuff that happens a player couldn't fall too far behind so there is still a rush to things but you don't have to always get it right the first time or reset time (4th wall breaking).

I'd like to see much fewer but deeper quests. Save the epics for after 1.0 and establish flavor and baseline first. Don't make it so big becoming familiar enough to contribute is impossible/undesirable.

If there were only a dozen or so mission per story line getting them bug free and complete would be much easier.

I've seen a lot of projects hit this point and forget that just because you can add content doesn't mean you should. People work on what works but then you end up with way too much of that.

Right now it's possible to do major balance changes manually, it might take a few days to make sure everything has it's values changed correctly but that's quickly spinning out of control.

I haven't seen much of the story line despite pursuing it because it's confusing and obtuse (to be taken as constructive criticism, free is free this isn't complaining, content creators prerogative) so I can't start biting into fixing it not knowing if I'll be messing up the plot later since there is no plot bible.

Forking might be easier, a minimalist version for fast development. Dozen stars are so for humans space 4 to 6 for alien. Cut down ships to a minimum, just their most representative craft. Every mission would be a demo as well doubling as a copy paste library, so just once each. Get a solid seed then extend.

A huge side benefits is being able to test out balance changes without taking on a massive task. I like the idea of a Starship Troopers style space combat with huge hull values but changing EVERY ship to test the theory is beyond me. Changing three dozen entries for every major system, can be done in an afternoon with find and replace. I'm more of a combat should take a minute not 10 seconds kind guy.

Mini Sky could also be the try it demo. Why can't a free game come with a demo? People with restricted bandwidth and data-caps would probably be interested.

I'm so bad about going off topic, sorry about that. Disregard anything that sounded stupid please. Thinking out loud because I want to do my own thing and there is so much to choose from. And you know what they say, if you want it done right...

**Comment by Bladewood** [comment]
Well of course, you can do that, but it would take much more time and effort to code and write all the branches. Isn't that what we're trying to do, anyway?

**Comment by jafdy** [comment]
So your proposing that we have less story and if people wan't to try all the story they need to fail some missions?
The biggest single complaint (not including requests for multiplayer) is that there isn't enough story or they can't find the story. I fell that having less story will not help much with that.
If you wan't to write more ways of branching the story (and a branch after a lost battle would be cool) then you can always write it your self.

**Comment by Amazinite** [comment]
There are really very few places that you can really get a second chance without the entire storyline changing from what it currently is. FW Second Chance is the only place I can think of right now where you can fail a mission and it gives you that second chance to prove yourself, but that's before war breaks out and it doesn't seem to be that high a failure. 

During the war, on the other hand, where can you fail and have a serious shot at a second chance that doesn't involve a massive story rewrite? The only way you can fail battles is if you die, so that ends anyways. If there's a mission where you have an important escort, that escort normally has a major character in it and therefore it would take a lot of writing to have a second chance after losing it. The only place I can really think of off the top of my head that could have a second chance to it is when you're told to rush to Rastaban ASAP to defend Dancer. Failing to get there in time could mean that the Navy takes the system, and therefore the Free Worlds have more difficulty traveling through the region.

**Comment by jafdy** [comment]
Had someone else abort a key mission by accident.

**Comment by AlbertNewton** [comment]
Yes, I accidentally aborted one of the Remnant missions.

**Comment by Amazinite** [comment]
There is already a dialog that appear when you attempt to abort a mission that asks you if you're sure. We could add a new mission node that adds more text to this abort dialog if it is present and give all the FW intro missions this dialog, akin to how the missions after you've joined the FW have the extra dialog when you fail a mission.
![image](https://github.com/endless-sky/endless-sky/assets/17688683/090dbf9b-6b1e-463a-a7e6-81588891d52a)
https://github.com/endless-sky/endless-sky/blob/d12301477d6fc2a6231d90004ce4daed4ed5b5fd/source/MissionPanel.cpp#L291-L292

**Comment by Zireael07** [comment]
> We could add a new mission node that adds more text to this abort dialog if it is present 

Yes please. Currently a first-time player has no clue that some missions are more important than others

**Comment by slchorne** [comment]
Following through on this thread, especially since it relates to (FW Katya 3). This is a hard mission for a newbie to complete because there is no good tutorial/guide/help on how to explicitly hunt the drone. New users aren't taken on a training mission to target a ship, or how to board them. (e.g: get close, disable first, auto track or auto boarding). New users learn the simplicities of 'J' = Jump and 'L'= land. But this technique is more nuanced, even if you do succeed in the asteroid capture mission. 

If this is a critical node in the storyline, there needs to be a workaround, or alternate path to avoid a frustrated player just aborting the mission
